zipline/pipeline/engine.py

```diff
@@ -384,6 +384,20 @@ def _to_narrow(self, data, mask, dates, assets):
         If mask[date, asset] is True, then result.loc[(date, asset), colname]
         will contain the value of data[colname][date, asset].
         """
+        if not mask.any():
+            # Manually handle the empty DataFrame case. This is a workaround
```

Comment:
Might be worth noting that this also saves us the work of applying a known-empty mask to the array.